egs/multi_en/s5/local/chain/run_blstm_6h.sh

```diff
@@ -0,0 +1,188 @@
+# Changes made:
+#  - TODO
+###########################################################################################
+
+# based on run_tdnn_6h.sh
+
+set -e
+
+# configs for 'chain'
+stage=12
+train_stage=-10
+get_egs_stage=-10
+dir=exp/chain/blstm_6h
+decode_iter=
+decode_dir_affix=
+
+# training options
+num_epochs=4
+remove_egs=false
+common_egs_dir=
+affix=
+chunk_width=150
+chunk_left_context=40
+chunk_right_context=40
+
+# End configuration section.
+echo "$0 $@"  # Print the command line for logging
+
+. ./cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+dir=$dir${affix:+_$affix}
+train_set=train_nodup_sp
+ali_dir=exp/tri5a_ali_nodup
+treedir=exp/chain/tri6_tree_11000
+lang=data/lang_chain
+
+# The iVector-extraction and feature-dumping parts are the same as the standard
+# nnet3 setup, and you can skip them by setting "--stage 8" if you have already
+# run those things.
+local/nnet3/run_ivector_common.sh --stage $stage \
+  --speed-perturb true \
+  --generate-alignments false || exit 1;
+
+if [ $stage -le 9 ]; then
+  # Get the alignments as lattices (gives the CTC training more freedom).
+  # use the same num-jobs as the alignments
+  nj=$(cat $ali_dir/num_jobs) || exit 1;
+  steps/align_fmllr_lats.sh --nj $nj --cmd "$train_cmd" data/$train_set \
+    data/lang exp/tri5a exp/tri5a_lats_nodup_sp
+  rm exp/tri5a_lats_nodup_sp/fsts.*.gz # save space
+fi
+
+if [ $stage -le 10 ]; then
+  # Create a version of the lang/ directory that has one state per phone in the
+  # topo file. [note, it really has two states.. the first one is only repeated
+  # once, the second one has zero or more repeats.]
+  rm -rf $lang
+  cp -r data/lang $lang
+  silphonelist=$(cat $lang/phones/silence.csl) || exit 1;
+  nonsilphonelist=$(cat $lang/phones/nonsilence.csl) || exit 1;
+  # Use our special topology... note that later on may have to tune this
+  # topology.
+  steps/nnet3/chain/gen_topo.py $nonsilphonelist $silphonelist >$lang/topo
+fi
+
+if [ $stage -le 11 ]; then
+  # Build a tree using our new topology.
+  steps/nnet3/chain/build_tree.sh --frame-subsampling-factor 3 \
+      --leftmost-questions-truncate -1 \
+      --cmd "$train_cmd" 11000 data/$train_set $lang $ali_dir $treedir
+fi
+
+if [ $stage -le 12 ]; then
+  echo "$0: creating neural net configs";
+
+  steps/nnet3/lstm/make_configs.py  \
+    --feat-dir data/${train_set}_hires \
+    --ivector-dir exp/nnet3/ivectors_${train_set} \
+    --tree-dir $treedir \
+    --splice-indexes="-2,-1,0,1,2 0 0" \
+    --lstm-delay=" [-3,3] [-3,3] [-3,3] " \
+    --xent-regularize 0.1 \
+    --include-log-softmax false \
+    --num-lstm-layers 3 \
+    --cell-dim 1024 \
+    --hidden-dim 1024 \
+    --recurrent-projection-dim 256 \
```

Comment:
I am using 256 + 256.


在 2016/6/28 1:59, Vijayaditya Peddinti 写道:
>
> In egs/multi_en/s5/local/chain/run_blstm_6h.sh 
> <https://github.com/kaldi-asr/kaldi/pull/771#discussion_r68624791>:
>
> > +
> > +if [ $stage -le 12 ]; then
> > +  echo "$0: creating neural net configs";
> > +
> > +  steps/nnet3/lstm/make_configs.py  \
> > +    --feat-dir data/${train_set}_hires \
> > +    --ivector-dir exp/nnet3/ivectors_${train_set} \
> > +    --tree-dir $treedir \
> > +    --splice-indexes="-2,-1,0,1,2 0 0" \
> > +    --lstm-delay=" [-3,3] [-3,3] [-3,3] " \
> > +    --xent-regularize 0.1 \
> > +    --include-log-softmax false \
> > +    --num-lstm-layers 3 \
> > +    --cell-dim 1024 \
> > +    --hidden-dim 1024 \
> > +    --recurrent-projection-dim 256 \
>
> When you start tuning your BLSTM system you could probably increase 
> the projection dims to 512, as there is considerable amount of 
> training data in this recipe.
>
> @naxingyu <https://github.com/naxingyu> What are you using in your 
> updated BLSTM+chain recipes for fisher+swbd ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/kaldi-asr/kaldi/pull/771/files/d13323180b98a83622b0cb193a3a5cd042468609#r68624791>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ADKpxL4SSK6QOamE6SqLimRFWT4yYiXxks5qQA9rgaJpZM4IZ3x7>.
>